### PR TITLE
Implement recent searches

### PR DESCRIPTION
**Pull request by thagikura**
- Add Repository and Dao for recent searches
    - Add recent searches body in the Search UI
    - Add relevant tests
    
Screenshots:

### CODE REVIEW

**Comment by thagikura** [comment]
#668 should be merged first because this PR is created on top of it.

File: feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt [hunk 1]
```diff
@@ -16,15 +16,85 @@
...
-        // TODO: Pass it to UseCase
-        println("New search query is '$searchQuery'")
+    val searchQuery = savedStateHandle.getStateFlow("searchQuery", "")
+
+    val searchResultUiState: StateFlow<SearchResultUiState> =
+        searchQuery.flatMapLatest { query ->
+            if (query.length < 2) {
+                flowOf(SearchResultUiState.EmptyQuery)
+            } else {
+                try {
```
Comment (hoangchungk53qx1) [inline feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt:52]:
```suggestion
                              getSearchContentsUseCase(query)
                    .map {
                        SearchResultUiState.Success(
                            topics = it.topics,
                            newsResources = it.newsResources,
                        )
                    }.catch {
                        flowOf(LoadFailed)
                    }
```

**Review by hoangchungk53qx1 — COMMENTED** [review]
you can use .catch in flow

File: feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt (hunk 1 above)
Comment (thagikura) [reply feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt]: Done

**Comment by thagikura** [comment]
> you can use .catch in flow

Addressed. Thanks.

**Comment by thagikura** [comment]
Note to reviewers, 
@dturner @neelanshsahai 
#668 needs to be reviewed first because this PR is created on top of it. But

**Comment by thagikura** [comment]
Closing this in favor of #675 since I had to resolve a lot of conflicts with this PR.

**Pull request closed without merging**